presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/DatabaseShardManager.java

```diff
@@ -274,10 +276,22 @@ public void addColumn(long tableId, ColumnInfo column)
                 maxColumn(column.getColumnId()), columnType);
 
         try (Handle handle = dbi.open()) {
```

Comment:
Move this inside the loop so that we acquire a fresh connection each time. Otherwise, if we fail due to connection error, we'll keep failing on the same bad connection.